flink-ml-python/pyflink/ml/core/linalg.py

```diff
@@ -776,3 +777,15 @@ def _double_to_long_bits(value: float) -> int:
         value = float("nan")
     # pack double into 64 bits, then unpack as long int
     return struct.unpack("Q", struct.pack("d", value))[0]
+
+
+parent_from_java_type = typeinfo._from_java_type
+
+
+def _from_java_type(j_type_info: JavaObject):
+    if "GenericType<org.apache.flink.ml.linalg.DenseVector>" == str(j_type_info):
+        return DenseVectorTypeInfo()
+    return parent_from_java_type(j_type_info)
+
+
+typeinfo._from_java_type = _from_java_type
```

Comment:
Would it be better to put the global module changes in `flink-ml-python/pyflink/ml/__init__.py`?

Reply:
It might be a little complicated to import a project's own class (DenseVectorTypeInfo) in the project's `__init__.py`. The current best practice is to define the module change in the most related non-init file, like the change of `_from_java_type_wrapper` in `pyflink/ml/core/wrapper.py`.

Reply:
According to offline discussion, I'll merge the logic related to java type here into that defined in `wrapper.py`, and extract the common logic of functions into wrapper.py. In this case, all usages of java vector type would go through wrapper.py and use the modified version of `_from_java_type()`.